### ISSUE TITLE
[Bug]: There is no way to reset the pagination variables in GraphQL query

### ISSUE THREAD

**Issue by rohan-arthur** [appsmithorg/appsmith#16176]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://user-images.githubusercontent.com/94514895/185849311-b4057a1d-319f-4825-b924-377edbee516f.mov

### Steps To Reproduce

1. Create GraphQL query
2. Create some variables in the body
3. Select these variables in the pagination tab
4. Now go back to body and remove those variables (you might have decided against using pagination for the moment)
5. Go back to Pagination tab and as you can see in the video, you cannot remove the selected variables - you can still set the pagination option to `none`, but when you switch back to the previously selected option, these variables are still present and they cannot be removed.

### Version

Cloud

**Comment by sumitsum** [comment]
@AmanAgarwal041 assigning this to you since this looks like something that should be handled from he client side. Please let me know if it needs to be handled on server side.

**Comment by AmanAgarwal041** [comment]
@rohan-arthur @parth-appsmith we will be requiring design to add this functionality.

**Comment by parth-appsmith** [comment]
@AmanAgarwal041 from what I understand of this issue is that once the selected variables are removed from the query variable they still appear selected in the pagination with an error (Since those variables doesn't exists). 
I dont see where are designs required for this, shouldn't the interaction be as soon as the source of the variable doesn't exists then there is no point that they should appear in the dropdown. The dropdown in this case should fall back to default state (nothing is selected). 

Even providing a CTA to reset values is adding an extra step of friction to an experience which should happen automatically. 

CC: @rohan-arthur

**Comment by rohan-arthur** [comment]
@parth-appsmith you're right of course, and that's the bug we want to fix. However, the component itself can be better: suppose I select something in the drop-down, how can I un-select it again? This is currently not working. I imagine a "blank" value in the drop-down should work - would you please suggest a suitable existing component?

**Comment by rohan-arthur** [comment]
There are two scenarios we need to address:
1. allow users to have nothing selected, after they once select something?
2. where the dropdown is bound to a variable and the variable is deleted, what should happen?

As discussed today, let's confine this issue to scenario 1. Scenario 2 will be tackled in a separate issue. 

The solution should be similar to the action selector, in which I can reset my trigger action after selecting something first:
<img width="247" alt="Screenshot 2022-09-19 at 18 08 26" src="https://user-images.githubusercontent.com/94514895/191019320-44f12e6f-83a0-4eb0-9936-b69349af490b.png">

However, instead of `No action`, this dropdown should have the reset option as `None`, so that: 
1. On selecting this, the selection should effectively be cleared. 
2. This additional option should be available in all the dropdowns in the graphql pagination tab

**Comment by sumitsum** [comment]
6m usage of GraphQL: 341
Impacted % of users: 25 
## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 85 |
| Effort (months)  | 0.25 |

**Comment by rohan-arthur** [comment]
assigning [medium](https://miro.com/app/board/uXjVPuhukFg=/?messenger=slack&utm_source=slack&utm_medium=messenger&utm_campaign=notifications&utm_content=unfurled_chat_link&utm_term=messenger_analytics) priority to this
